core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts

```diff
@@ -47,14 +55,13 @@ export class DotAiImagePromptStore extends ComponentStore<DotAiImagePromptCompon
         this.state$,
         ({ status }) => status === ComponentStatus.LOADING
     );
+
+    readonly selectedImage$ = this.select(this.state$, ({ selectedImage }) => selectedImage);
```

Comment:
Could be `readonly selectedImage$ = this.select(this.state$, ({ galleryActiveIndex, images }) => galleryActiveIndex ? images[galleryActiveIndex]) : 0;`

With the index can get the selected image.

Reply:
not in this case, the selectedImage is the one that goes to the block editor, the array of images are the ones that are being generated by the user in the popup.